pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -195,7 +205,7 @@ let
     inherit kernelPatches randstructSeed extraMakeFlags extraMeta configfile;
     pos = builtins.unsafeGetAttrPos "version" args;
 
-    config = { CONFIG_MODULES = "y"; CONFIG_FW_LOADER = "m"; };
+    config = { CONFIG_MODULES = "y"; CONFIG_FW_LOADER = "m"; } // lib.optionalAttrs withRust { CONFIG_RUST = "y"; };
```

Comment:
Why do we need to do this for Rust, but not most other options?

Reply:
sadly this is a bit of a hack. `manualConfig` gets the kernel config passed as a text file (derivation) `configfile` and such can not do eval decisions based on this. the `config` attribute set is _supposed_ to be the nix attrset representation of the `configfile` contents, but I suppose because of performance reasons we only pass the options that are actually interpreted by manualConfig.
Anyways without this, manualConfig will not know for eval (specifying dependencies) if the config is rust-enabled.